File: client/lib/annoucements/pet.dart
import 'dart:typed_data';
import 'package:json_annotation/json_annotation.dart';

@JsonSerializable()
class Pet {
  Pet(
      {this.name = "",
      this.species = "",
      this.birthday,
      this.breed = "",
      this.description = "",
      this.photo});

  String name;
  String species;
  String breed;
  DateTime? birthday;
  String description;
  Uint8List? photo;
}

File: apis/DatabaseContextLibrary/Models/Pet.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace DatabaseContextLibrary.models
{
    public class Pet
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public Guid ShelterId { get; set; }
        public Shelter Shelter { get; set; }
        public string Name { get; set; }
        public string Species { get; set; }
        public string Breed { get; set; }
        public DateTime Birthday { get; set; }
        public string Description { get; set; }
        public byte[]? Photo { get; set; }
    }
}
